### PR TITLE
added basic carousel framework with progression dots

### PR DESCRIPTION
**Pull request by huai-codes**
### Description

Added basic Carousel framework as well as the dots for progression tracking, content still to be fixed

### QA

Just try swiping through each, should be pretty clean. Any feedback on formatting appreciated.

### Packages

Added react-native-snap-carousel for the Carousel and the progression dots (paginatino)

### Checklist

- [ ] Code compiles correctly
- [ ] Added comments
- [ ] Ran linter and made sure there were no errors
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: books-and-breakfast/app/MissionScreen.tsx
Comment (JackyZhang1219) [inline books-and-breakfast/app/MissionScreen.tsx:1]: Getting a ViewPropType error - not sure what the issue is

File: books-and-breakfast/app/MissionScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    title: 'Primary Strategies',
+    body: '"I belong in this academic community." "My ability and competence grow with my effort." "I can succeed at this." "This work has value for me."',
+  },
+  {
+    title: 'The Tutor Role',
+    body: "Friend, Cheerleader, Coach, Driver's Education Teacher, Learner - each plays a vital role in supporting the students.",
+  },
+  {
+    title: 'The Site Director Role',
+    body: "Daily Program Manager, Family and Student Connector, District 65 Partner, Student and Family Advocate - responsible for overseeing the program's daily operations and long-term success.",
```
Comment (JackyZhang1219) [inline books-and-breakfast/app/MissionScreen.tsx:41]: Can you add all the text from the figma

File: books-and-breakfast/app/MissionScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        sliderWidth={screenWidth}
+        itemWidth={screenWidth}
+        layout="default"
+        onSnapToItem={(index) => setActiveSlide(index)}
+      />
+      <Pagination
+        dotsLength={entries.length}
+        activeDotIndex={activeSlide}
+        containerStyle={styles.paginationContainer}
+        dotStyle={styles.dotStyle}
```
Comment (JackyZhang1219) [inline books-and-breakfast/app/MissionScreen.tsx:69]: is there a way to make the dots appear closer together like the figma mockup?

File: books-and-breakfast/app/MissionScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+interface Entry {
+  title: string;
+  subtitle?: string;
+  body: string;
+  link?: string;
+}
+
+const { width: screenWidth } = Dimensions.get('window');
+
+const entries: Entry[] = [
```
Comment (JackyZhang1219) [inline books-and-breakfast/app/MissionScreen.tsx:14]: it would prob be cleaner if you move this data to a separate file in the data folder

**Review by JackyZhang1219 — COMMENTED** [review]
Can you upload the data in firebase and fetch it if you have time?

File: books-and-breakfast/tsconfig.json [hunk 4]
```diff
@@ -7,5 +7,5 @@
       "@/*": ["src/*"]
     }
   },
-  "include": ["**/*.ts", "**/*.tsx", "babel.config.js"]
+  "include": ["**/*.ts", "**/*.tsx", "babel.config.js", "app/data/MissionData"]
```
Comment (JackyZhang1219) [inline books-and-breakfast/tsconfig.json:10]: Is this necessary?

**Comment by huai-codes** [comment]
oh my bad that was prettier i think…?

Harry Guan

File: books-and-breakfast/app/MissionScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  container: {
+    flex: 1,
+    backgroundColor: 'white', // Sets the background color for the entire screen
+  },
+  item: {
+    backgroundColor: 'white',
+    borderRadius: 5,
+    height: 500,
+    padding: 20,
+    marginLeft: 30,
```
Comment (JackyZhang1219) [inline books-and-breakfast/app/MissionScreen.tsx:65]: can you remove the margins and change padding to 30 just so it's consistent with the directions screen

File: books-and-breakfast/app/MissionScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+LogBox.ignoreLogs(['ViewPropTypes will be removed from React Native']);
+
+const { width: screenWidth } = Dimensions.get('window');
+
+const MissionScreen: React.FC = () => {
+  const [activeSlide, setActiveSlide] = useState(0); // State to track active slide index
+
+  const renderItem = ({ item }: { item: Entry }) => (
+    <View style={styles.item}>
+      <Text style={styles.title}>{item.title}</Text>
```
Comment (JackyZhang1219) [inline books-and-breakfast/app/MissionScreen.tsx:17]: can you add a little space between title and first subtitle

**Pull request merged**